OpenRA.Game/Activities/Activity.cs

```diff
@@ -234,20 +234,28 @@ public virtual bool Cancel(Actor self, bool keepQueue = false)
 			return true;
 		}
 
-		public virtual void Queue(Activity activity)
+		public virtual void Queue(Actor self, Activity activity, bool pretick = false)
 		{
 			if (NextInQueue != null)
-				NextInQueue.Queue(activity);
+				NextInQueue.Queue(self, activity);
 			else
+			{
 				NextInQueue = activity;
+				if (pretick)
+					NextInQueue = ActivityUtils.RunActivity(self, NextInQueue);
+			}
 		}
 
-		public virtual void QueueChild(Activity activity)
+		public virtual void QueueChild(Actor self, Activity activity, bool pretick = false)
 		{
 			if (ChildActivity != null)
-				ChildActivity.Queue(activity);
+				ChildActivity.Queue(self, activity);
 			else
+			{
 				ChildActivity = activity;
+				if (pretick)
```

Comment:
Style nit: this could be written more concisely as `ChildActivity = pretick ? ActivityUtils.RunActivity(self, activity) : activity;`

Reply:
Fixed.